HomeAutomationServer/Models/AppCache.cs

```diff
@@ -46,51 +51,56 @@ static public bool AddDeviceBlob_DEBUG(JObject blob)
 
 		/*************** END DEBUG MODE METHOD *************/
 
-        static public bool AddDeviceBlob(JObject blob)
-        {
-            deviceBlobs[(string)blob["deviceID"]] = blob;
-
-            if (deviceBlobs[(string)blob["deviceID"]] == blob)
-            {
-                try
-                {
-                    AmazonSimpleNotificationServiceClient snsClient = new AmazonSimpleNotificationServiceClient("AKIAJM2E3LGZHJYGFSQQ", "p3Qi8DAXj+XHAH+ny7HrlRyleBs5V5DJv77zKK3T", Amazon.RegionEndpoint.USEast1);
-                    snsClient.Publish("arn:aws:sns:us-east-1:336632281456:MyTopic", "New Device Updates");
-                }
-
-                catch (Exception ex)
-                {
-                    LogFile.AddLog("AppCache -- Could not send Push Notification: " + ex.Message + "\n");
-                    return false;
-                }
-
-                return true;
-            }
-
-            return false;
-        }
-
-        static public JToken GetDeviceBlob(string deviceID)
-        {
-            JToken blob = deviceBlobs[deviceID];
-            deviceBlobs[deviceID].Remove();
-            return blob;
-        }
-
-        static public JArray GetAllBlobs()
-        {
-            JArray blobs = deviceBlobs;
-            deviceBlobs.RemoveAll();
-            return blobs;
-        }
-
-        static public int GetBlobCount()
-        {
-            return deviceBlobs.Count;
-        }
-
-        ////////////////////////////////////////////////////////////////////////////////////////
-        //
-        // other blob methods
-    }
+		static public bool AddDeviceBlob(JObject blob)
+		{
+			string blob_string = blob.ToString ();
+			Device dev = Interfaces.DeserializeDevice(blob_string, null, null, new TimeFrame()); // to convert the JSON blob to an actual Device object.
+			string devID = (string)blob["deviceID"];
+
+			deviceBlobs.Add(devID, dev);
+
+			if (deviceBlobs.Contains(dev)) {
+				try
+				{
+					AmazonSimpleNotificationServiceClient snsClient = new AmazonSimpleNotificationServiceClient("AKIAJM2E3LGZHJYGFSQQ", "p3Qi8DAXj+XHAH+ny7HrlRyleBs5V5DJv77zKK3T", Amazon.RegionEndpoint.USEast1);
+					snsClient.Publish("arn:aws:sns:us-east-1:336632281456:MyTopic", "New Device Updates");
+				}
+
+				catch (Exception ex)
+				{
+					LogFile.AddLog("AppCache -- Could not send Push Notification: " + ex.Message + "\n");
+					return false;
+				}
+
+				return true;
+			}
+
+			return false;
+		}
+
+		static public JToken GetDeviceBlob(string deviceID)
```

Comment:
If the string is a JSON object, it should be parsed into an actual object, otherwise any change in formatting will ruin the Dictionary. If the string is a JSON object that matches how a Device needs to be identified according to the system design, you should parse it into a FullID object. If this string is only the device ID portion of a FullID (e.g. missing a room and house ID), then this function is incomplete, because there isn't enough data to know which Device is being referenced.